pnet_sys/src/unix.rs

```diff
@@ -28,7 +28,46 @@ pub mod public {
     pub const IP_HDRINCL: libc::c_int = libc::IP_HDRINCL;
 
     pub const IFF_LOOPBACK: libc::c_int = libc::IFF_LOOPBACK;
+    
+    #[cfg(target_os = "macos")]
+    mod device_flags {
+        use super::libc;
+        // XNU
+        /// https://github.com/aosm/xnu/blob/master/bsd/net/if.h#L140-L156
+        pub const IFF_UP: libc::c_int          = 0x1;       //  interface is up
+        pub const IFF_BROADCAST: libc::c_int   = 0x2;       //  broadcast address valid
+        pub const IFF_DEBUG: libc::c_int       = 0x4;       //  turn on debugging
+        // pub const IFF_LOOPBACK: libc::c_int    = 0x8;       //  is a loopback net
+        pub const IFF_POINTOPOINT: libc::c_int = 0x10;      //  interface is point-to-point link
+        pub const IFF_NOTRAILERS: libc::c_int  = 0x20;      //  obsolete: avoid use of trailers
+        pub const IFF_RUNNING: libc::c_int     = 0x40;      //  resources allocated
+        pub const IFF_NOARP: libc::c_int       = 0x80;      //  no address resolution protocol
+        pub const IFF_PROMISC: libc::c_int     = 0x100;     //  receive all packets
+        pub const IFF_ALLMULTI: libc::c_int    = 0x200;     //  receive all multicast packets
+        pub const IFF_OACTIVE: libc::c_int     = 0x400;     //  transmission in progress
+        pub const IFF_SIMPLEX: libc::c_int     = 0x800;     //  can't hear own transmissions
+        pub const IFF_LINK0: libc::c_int       = 0x1000;    //  per link layer defined bit
+        pub const IFF_LINK1: libc::c_int       = 0x2000;    //  per link layer defined bit
+        pub const IFF_LINK2: libc::c_int       = 0x4000;    //  per link layer defined bit
+        pub const IFF_ALTPHYS: libc::c_int     = IFF_LINK2; //  use alternate physical connection
+        pub const IFF_MULTICAST: libc::c_int   = 0x8000;    //  supports multicast
+    }
+
+    #[cfg(target_os = "linux")]
+    mod device_flags {
+        use super::libc;
+        // Linux
+        // https://github.com/torvalds/linux/blob/master/include/uapi/linux/if.h#L83-L100
+        // http://man7.org/linux/man-pages/man7/netdevice.7.html
+        pub use libc::{IFF_UP, IFF_BROADCAST, IFF_DEBUG, IFF_POINTOPOINT, 
```

Comment:
Have you made a PR to `libc` for the macOS constants? Since they exist on Linux, their equivalents should proably be added for macOS too.

Reply:
Getting it merged upstream would be gold.

Reply:
https://github.com/rust-lang/libc/pull/839

Reply:
Now when they exist upstream, after a release happens, then this would not need to be separate `device_flags` modules for `linux` and `macos`. The unix part of libpnet could just use `libc::IFF_*` directly.